EIPS/eip-838.md

```diff
@@ -0,0 +1,96 @@
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+## Specification
+
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+
+To conform to this specification, compilers producing JSON ABIs SHOULD include error declarations alongside functions and events. Each error object MUST contain the keys `name` (string) and `arguments` (same types as the function’s `inputs` list). The value of `type` MUST be "error".
+Example:
+
+```json
+{
+  "type": "error",
+  "name": "InsufficientBalance",
+  "arguments": [
+    {
+      "name": "amount",
+      "type": "uint256"
+    }
+  ]
+}
+```
+
+A selector for this error can be computed from its signature (`InsufficientBalance(uint256)` for the example above) in the same way that it's currently done for public functions and events. Any arguments for the error together with the selector are ABI encoded in the same way as call data for functions.
+
+A high-level language like Solidity can then implement a syntax like this:
+
+```solidity
+contract MyToken {
+  error InsufficientFunds(uint256 amount);
+
+  function transfer(address _to, uint256 _amount) {
+    if (balances[msg.sender] <= _amount) {
+       revert InsufficientFunds(_amount);
+    }
+  }
+}
+```
+
+## Rationale
+
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+
+This specific encoding was choosen because it is the same one Solidity uses at this moment for REVERTs. This PR just proposes to allow for user-defined errro ABIs instead of hard-coding `Error(string)`.
+
+In terms of previous work - there is one other [EIP 207](https://github.com/ethereum/EIPs/pull/207). It proposes to use **CBOR** encoding. In my oponion - there is no need to introduce another encoding scheme because the ABI encoding is flexible fast cheap and supported already.
```

Comment:
```suggestion
In terms of previous work - there is one other [EIP 207](https://github.com/ethereum/EIPs/pull/207). It proposes to use **CBOR** encoding. In my oponion - there is no need to introduce another encoding scheme because the ABI encoding is flexible, fast, cheap and already supported.
```